### ISSUE TITLE
about depth loss increasing.

### ISSUE THREAD

**Issue by rockywind** [nerfstudio-project/nerfstudio#1447]
Hi,
When I add the use the `depth-nerfacto`,  the depth loss and intervel loss increased.\
The train script is below.
````
export CUDA_VISIBLE_DEVICES=0
ns-train depth-nerfacto  --viewer.websocket-port=7009 \
                --machine.num-gpus 1 \
                --experiment-name kitti_00_200_depth \
                --data data/kitti/kitti_00_200_depth \
                --vis tensorboard 
````
The detailed log is below.
![image](https://user-images.githubusercontent.com/14838781/219670436-3a04165c-3c86-4b55-93e0-bc9f473c5235.png)
![image](https://user-images.githubusercontent.com/14838781/219672319-0c1ef6da-39bc-4de7-9cfc-e68f511131a9.png)
The predicted depth map
![image](https://user-images.githubusercontent.com/14838781/219672553-29453f0c-810a-4bb9-9e6f-b34fba39ce05.png)
The render image
![image](https://user-images.githubusercontent.com/14838781/219672835-c8aa3a13-f267-4e7b-85d5-b9c2fd520a5d.png)

**Comment by pixel-alex** [comment]
It looks interesting! Could you please share how you preprocess the kitti data? Thanks!

**Comment by AIBUWAN** [comment]
@rockywind I have the same problem, please tell me how you solved it.

**Comment by ayushjain1144** [comment]
I am observing similar issue on scannet

**Comment by cxzhou35** [comment]
> Hi, When I add the use the `depth-nerfacto`, the depth loss and intervel loss increased. The train script is below.
> 
> ```
> export CUDA_VISIBLE_DEVICES=0
> ns-train depth-nerfacto  --viewer.websocket-port=7009 \
>                 --machine.num-gpus 1 \
>                 --experiment-name kitti_00_200_depth \
>                 --data data/kitti/kitti_00_200_depth \
>                 --vis tensorboard 
> ```
> 
> The detailed log is below. ![image](https://user-images.githubusercontent.com/14838781/219670436-3a04165c-3c86-4b55-93e0-bc9f473c5235.png) ![image](https://user-images.githubusercontent.com/14838781/219672319-0c1ef6da-39bc-4de7-9cfc-e68f511131a9.png) The predicted depth map ![image](https://user-images.githubusercontent.com/14838781/219672553-29453f0c-810a-4bb9-9e6f-b34fba39ce05.png) The render image ![image](https://user-images.githubusercontent.com/14838781/219672835-c8aa3a13-f267-4e7b-85d5-b9c2fd520a5d.png)

Same problem as yours, have you found any reasons or solutions?

**Comment by ethanweber** [comment]
Is the depth loss for eval being computed on your sparse depth input? If so, there are a lot of zeroes for the invalid regions.

**Comment by ayushjain1144** [comment]
Thank you, Ethan.

The depth is not too sparse I think, I am using depth maps from scannet, which has holes where the kinect sensor didn't record valid depth. The depth loss is masked out at those invalid locations. 

I checked the min/max of the scannet depth (which is gt) and the nerf predicted depth. I am noticing that they have same min/Max values i.e. between (0,9) but the sensor depth is usually between 0-2 with some outliers > 2, while the nerf depth has lots of depths > 2. My guess is that it is being sensitive to these outliers, and if I mask out the outliers from ground-truth, it might work.  I am going to try it now and report back how it went.

Let me know what you think, and if you think there is a better way to deal with outliers, let me know.

**Comment by ethanweber** [comment]
Does the depth of the NeRF look reasonable when you don't train the depth-nerfacto model and instead just train a normal nerfacto model -- perhaps with more paramaters (nerfacto-huge, etc.)? I guess the outliers could cause some issues, as you're suggesting. If the NeRF depends on the depth to get good results, then it's probably just a hard setting where there isn't enough parallax in your training data. Maybe worth switching to a surface-based method in that case.

Also maybe you could try relative depth supervision (i.e., sparse nerf). It defeats the purpose of having exact metric depth, but it could be helpful to not hurt the NeRF from optimizing it's photometric objective.

**Comment by ayushjain1144** [comment]
I did try training a nerf model without depth supervision -- it is the same architecture and parameters but I zero out the depth loss. The corresponding wandb is here: https://wandb.ai/ayushjain1144/nerfstudio-project/runs/9leiok9q?workspace=user-ayushjain1144. I think the depth maps look much worse compared to with depth supervision.

**Comment by ayushjain1144** [comment]
@ethanweber I studied your suggestion of training nerf model without depth supervision in more detail -- I find the predicted depth without any supervision is at very different scale than my "ground-truth" which might point to some convention / normalization differences. 

I am attaching below the precise numbers I see: you will notice that the predicted depth (after scaling back to original depth space) is very different than ground truth. The depths are z-depths i.e. distance from camera plane and I am using the default is_euclidean_depth: False

<img width="1006" alt="Screenshot 2023-10-26 at 12 00 17 PM" src="https://github.com/nerfstudio-project/nerfstudio/assets/28894174/db3a0099-8c21-41a0-b2f0-7c51851993c7">

Any help would be greatly appreciated! Thank you!
